tools/ds-identify

```diff
@@ -154,6 +159,8 @@ debug() {
     shift
     [ "$lvl" -gt "${DI_DEBUG_LEVEL}" ] && return
 
+    [ "$DI_LOG" = "" ] && DI_LOG="stderr"
```

Comment:
make sure the `debug()` function works, even if someone forgot to `set_run_path()`, or `DI_LOG`.

Reply:
Mentally walking through this:
1. call to `debug` before `set_run_path` causes `DI_LOG` set to `stderr`, `_DI_LOGGED` set to `stderr`, file descriptor 3 redirects to file descriptor 2
2. `set_run_path` redefines `DI_LOG` to `whatever/cloud-init.log`
3. subsequent call to `debug` sees that `"$_DI_LOGGED" != "$DI_LOG"`, re-opens fd 3 pointing to `whatever/cloud-init.log`, reassigns `_DI_LOGGED` to `$DI_LOG`
4. subsequent calls to `debug` use fd 3 which points to the right file

Looks good to me!